File: TeeNew/Builder2007/OpenGL_Editor.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "OpenGL_Editor.h"
#include "TeeGLEditor.hpp" // <<< IMPORTANT !!!!
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "Bar3D"
#pragma link "TeeEdit"
#pragma link "TeeOpenGL"
#pragma link "TeeTools"
#pragma resource "*.dfm"
TOpenGLEditorForm *OpenGLEditorForm;
//---------------------------------------------------------------------------
__fastcall TOpenGLEditorForm::TOpenGLEditorForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TOpenGLEditorForm::CheckBox1Click(TObject *Sender)
{
  Screen->Cursor = crHourGlass;
  try
  {
    TeeOpenGL1->Active = CheckBox1->Checked;
  }
  __finally
  {
    Screen->Cursor = crDefault;
  }
}
//---------------------------------------------------------------------------

void __fastcall TOpenGLEditorForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(6);
}
//---------------------------------------------------------------------------

void __fastcall TOpenGLEditorForm::Button1Click(TObject *Sender)
{
  ChartEditor1->Execute();
}
//---------------------------------------------------------------------------

File: TeeNew/OpenGL_Editor.pas
unit OpenGL_Editor;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, TeePyramid, TeeProcs,
  Chart, TeeEdit, Series, Bar3D, TeeOpenGL, TeeTools;

type
  TOpenGLEditorForm = class(TBaseForm)
    CheckBox1: TCheckBox;
    TeeOpenGL1: TTeeOpenGL;
    Series1: TBar3DSeries;
    Button1: TButton;
    ChartEditor1: TChartEditor;
    ChartTool1: TRotateTool;
    procedure CheckBox1Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

{$IFDEF WINDOWS}
Uses TeeGLEditor;  { <-- important to show the OpenGL editor at runtime !!! }
{$ENDIF}

procedure TOpenGLEditorForm.CheckBox1Click(Sender: TObject);
begin
  Screen.Cursor:=crHourGlass;
  try
    TeeOpenGL1.Active:=CheckBox1.Checked;
  finally
    Screen.Cursor:=crDefault;
  end;
end;

procedure TOpenGLEditorForm.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(6);
end;

procedure TOpenGLEditorForm.Button1Click(Sender: TObject);
begin
  ChartEditor1.Execute;
end;

initialization
  RegisterClass(TOpenGLEditorForm);
end.
